File: src/main/java/FileHandler.java
import java.io.File;

public class FileHandler implements FileAble{
    private final String pathToFile;
    private File[] listOfFiles;

    public FileHandler(String pathToFile) {
        this.pathToFile = pathToFile;
        initialize();
    }

    private void initialize(){
        File folder = new File(pathToFile);
        this.listOfFiles = folder.listFiles();
    }

    @Override
    public File[] getListOfFiles() {
        return listOfFiles;
    }

    @Override
    public String getFileName(int i){
        return pathToFile+listOfFiles[i].getName();
    }
}

File: src/SampleLibrary/SampleLibrary/FileHandler.cs
using SampleLibrary.Interfaces;

namespace SampleLibrary;

public class FileHandler : IFileAble
{
    private string[] _listOfFiles;

    public void LoadFile(string pathToFile)
    {
        _listOfFiles = Directory.GetFiles(pathToFile);
    }

    public string GetFileName(int numberOfFile)
    {
        return _listOfFiles[numberOfFile];
    }

    public string[] GetListOfFiles()
    {
        return _listOfFiles;
    }
}